nixos/modules/services/misc/godns.nix

```diff
@@ -0,0 +1,97 @@
+{ pkgs, lib, config, options, ... }:
+
+with lib;
+
+let
+  cfg = config.services.godns;
+
+  settingsFormat = pkgs.formats.json {};
+  settingsFile = settingsFormat.generate "godns.json" ({
+    ip_url = "https://ip4.seeip.org";
+    ipv6_url = "https://ip6.seeip.org";
+    ip_type = "IPv4";
+    interval = 300;
+    resolver = "8.8.8.8";
+  } // cfg.settings);
+
+in
+{
+  options = {
+    services.godns = {
+      enable = mkEnableOption "GoDNS";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.godns;
+        defaultText = literalExpression "pkgs.godns";
+        description = "GoDNS package to use";
+      };
+
+      settings = mkOption {
```

Comment:
Please do what is done in https://github.com/NixOS/nixpkgs/blob/39369610f81a6d6043612fd7bc409bbef623081c/nixos/modules/services/audio/navidrome.nix#L16